kernel/src/transaction/mod.rs

```diff
@@ -187,15 +187,32 @@ impl Transaction {
                 dup.app_id
             )));
         }
+        // Step 1: Generate SetTransaction actions
         let set_transaction_actions = self
             .set_transactions
             .clone()
             .into_iter()
             .map(|txn| txn.into_engine_data(get_log_txn_schema().clone(), engine));
 
-        // Step 2: Construct commit info and initialize the action iterator
+        // Step 2: Construct commit info with ICT if enabled
+        let in_commit_timestamp = if self
+            .read_snapshot
+            .table_configuration()
+            .is_in_commit_timestamps_enabled()
+        {
+            // Get the last commit timestamp and generate ICT timestamp
+            let last_commit_timestamp = self.read_snapshot.get_in_commit_timestamp(engine)?;
+            Some(generate_ict_timestamp(
+                self.commit_timestamp,
+                last_commit_timestamp,
+            ))
+        } else {
+            None
+        };
```

Comment:
Yeah, rust likes things a little more functional, mostly just avoiding cases like:
```rust
else { None }
```

The [then](https://doc.rust-lang.org/std/primitive.bool.html#method.then) method on `bool` is for just this, so we would do:


```suggestion
        let in_commit_timestamp = self
            .read_snapshot
            .table_configuration()
            .is_in_commit_timestamps_enabled().then(|| {
            // Get the last commit timestamp and generate ICT timestamp
            let last_commit_timestamp = self.read_snapshot.get_in_commit_timestamp(engine)?;
            generate_ict_timestamp(
                self.commit_timestamp,
                last_commit_timestamp,
            )
        });
```

which saves quite a few lines of code